File: leetcode/2_two_pointers/2109_Adding_Spaces_to_a_String.go
import "strings"

func addSpaces(s string, spaces []int) string {
	i, j := 0, 0
	res := make([]string, 0)

	for i < len(s) && j < len(spaces) {
		if i < spaces[j] {
			res = append(res, s[i:i+1])
			i++
		} else {
			res = append(res, " ")
			j++
		}
	}
	if i < len(s) {
		res = append(res, s[i:])
	}

	return strings.Join(res, "")
}

File: leetcode/2_two_pointers/2109_Adding_Spaces_to_a_String.py
def addSpaces(s: str, spaces: list[int]) -> str:
    i, j = 0, 0
    res = []

    while i < len(s) and j < len(spaces):
        if i < spaces[j]:
            res.append(s[i])
            i += 1
        else:
            res.append(" ")
            j += 1

    if i < len(s):
        res.append(s[i:])

    return "".join(res)


if __name__ == '__main__':
    print(addSpaces("EnjoyYourCoffee", [5, 9]))
    print(addSpaces("LeetcodeHelpsMeLearn", [8, 13, 15]))
